dgl/src/SVG.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * DISTRHO Plugin Framework (DPF)
+ * Copyright (C) 2012-2019 Filipe Coelho <falktx@falktx.com>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any purpose with
+ * or without fee is hereby granted, provided that the above copyright notice and this
+ * permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
+ * TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
+ * NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
+ * DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
+ * IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "SVG.hpp"
+
+#define NANOSVG_IMPLEMENTATION
+#include "nanosvg/nanosvg.h"
+
+#define NANOSVGRAST_IMPLEMENTATION
+#include "nanosvg/nanosvgrast.h"
+
+
+START_NAMESPACE_DGL
+
+SVG::SVG()
+    : fSize(),
+      fRGBAData(nullptr)
+{
+}
+
+void SVG::loadFromMemory(const char* const rawData, const uint dataSize, const float scaling) noexcept
+{
+    DISTRHO_SAFE_ASSERT_RETURN(rawData != nullptr, )
+    DISTRHO_SAFE_ASSERT_RETURN(scaling > 0.0f, )
+
+    if (fRGBAData != nullptr)
+    {
+        free(fRGBAData);
+        fRGBAData = nullptr;
+    }
+
+    NSVGrasterizer* rasterizer = nsvgCreateRasterizer();
+
+    // nsvgParse modifies the input data, so we must use a temporary buffer
+    const size_t bufferSize = dataSize * 4;
+    
+    char tmpBuffer[bufferSize];
```

Comment:
this might segfault if stack size becomes too big.
as this is not a RT operation, allocating memory seems safer